wire-runtime/src/main/java/com/squareup/wire/RuntimeFieldBinding.java

```diff
@@ -105,6 +105,12 @@ private RuntimeFieldBinding(int tag, String name, Message.Datatype datatype,
     this.messageField = messageField;
     this.builderField = builderField;
     this.builderMethod = builderMethod;
+
+    if (redacted && label == Message.Label.REQUIRED) {
```

Comment:
There’s 2 reasons to use redacted:

 * To obscure values in `toString()`
 * To strip fields

Presumably redacted+required is safe for the first case?